src/Cache_Command.php

```diff
@@ -406,4 +409,168 @@ public function flush_group( $args, $assoc_args ) {
+		list($key) = $args;
+		$group     = Utils\get_flag_value( $assoc_args, 'group' );
+
+		$value = wp_cache_get( $key, $group );
+
+		if ( false === $value ) {
+			WP_CLI::halt( 1 );
+		}
+
+		$key_path = array_map(
+			function ( $key ) {
+				if ( is_numeric( $key ) && ( (string) intval( $key ) === $key ) ) {
+					return (int) $key;
+				}
+				return $key;
+			},
+			array_slice( $args, 1 )
+		);
+
+		$traverser = new RecursiveDataStructureTraverser( $value );
+
+		try {
+			$value = $traverser->get( $key_path );
+		} catch ( \Exception $e ) {
+			die( 1 );
+		}
+
+		WP_CLI::print_value( $value, $assoc_args );
+	}
+
+	/**
+	 * Update a nested value from the cache.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <action>
+	 * : Patch action to perform.
+	 * ---
+	 * options:
+	 *   - insert
+	 *   - update
+	 *   - delete
+	 * ---
+	 *
+	 * <key>
+	 * : Cache key.
+	 *
+	 * <key-path>...
+	 * : The name(s) of the keys within the value to locate the value to pluck.
+	 *
+	 * [<value>]
+	 * : The new value. If omitted, the value is read from STDIN.
+	 *
+	 * [--group=<group>]
+	 * : Method for grouping data within the cache which allows the same key to be used across groups.
+	 * ---
+	 * default: default
+	 * ---
+	 *
+	 * [--expiration=<expiration>]
+	 *  : Define how long to keep the value, in seconds. `0` means as long as possible.
+	 *  ---
+	 *  default: 0
+	 *  ---
+	 *
+	 * [--format=<format>]
+	 * : The serialization format for the value.
+	 * ---
+	 * default: plaintext
+	 * options:
+	 *   - plaintext
+	 *   - json
+	 * ---
+	 */
+	public function patch( $args, $assoc_args ) {
+		list( $action, $key ) = $args;
+		$group                = Utils\get_flag_value( $assoc_args, 'group' );
+		$expiration           = Utils\get_flag_value( $assoc_args, 'expiration' );
+
+		$key_path = array_map(
+			function ( $key ) {
+				if ( is_numeric( $key ) && ( (string) intval( $key ) === $key ) ) {
+					return (int) $key;
+				}
+
+				return $key;
+			},
+			array_slice( $args, 2 )
+		);
+
+		if ( 'delete' === $action ) {
+			$patch_value = null;
+		} else {
+			$stdin_value = WP_CLI\Cache\Utils::has_stdin()
+				? trim( WP_CLI::get_value_from_arg_or_stdin( $args, -1 ) )
+				: null;
+
+			if ( ! empty( $stdin_value ) ) {
+				$patch_value = WP_CLI::read_value( $stdin_value, $assoc_args );
+			} elseif ( count( $key_path ) > 1 ) {
+				$patch_value = WP_CLI::read_value( array_pop( $key_path ), $assoc_args );
+			} else {
+				$patch_value = null;
+			}
+
+			if ( null === $patch_value ) {
+				WP_CLI::error( 'Please provide value to update.' );
+			}
+		}
+
+		/* Need to make a copy of $current_value here as it is modified by reference */
+		$old_value     = wp_cache_get( $key, $group );
+		$current_value = $old_value;
+		if ( is_object( $old_value ) ) {
+			$current_value = clone $old_value;
+		}
+
+		$traverser = new RecursiveDataStructureTraverser( $current_value );
+
+		try {
+			$traverser->$action( $key_path, $patch_value );
```

Comment:
How does the new cache value get written to a persistent object cache?

Reply:
As for the transient command, the cache value is updated later in the elseif condition with the `wp_cache_set` function.